### ISSUE TITLE
Upgrade fmtlib/fmt & gabime/spdlog dependencies

### ISSUE THREAD

**Issue by moderation** [envoyproxy/envoy#4744]
*Title*: Upgrade fmtlib/fmt & gabime/spdlog dependencies

*Description*:
`fmt` and `spdlog` have been recently been updated and have undergone significant changes. `fmt` had a major version change to 5.0.0 ([changelog](https://github.com/fmtlib/fmt/releases/tag/5.0.0)) and was released in May 2018. The layout of the library changed
> Moved fmt/*.h to include/fmt/*.h to prevent irrelevant files and directories appearing on the include search paths when fmt is used as a subproject and moved source files to the src directory.

I was able to have the new layout recognized by changing https://github.com/envoyproxy/envoy/blob/master/bazel/external/fmtlib.BUILD to:
```
cc_library(
    name = "fmtlib",
    srcs = glob([
        "src/*.cc"
    ]),
    hdrs = glob([
        "include/fmt/*.h",
    ]),
    defines = ["FMT_HEADER_ONLY"],
    includes = ["include"],
    visibility = ["//visibility:public"],
)
```
However this then generates many compilation issues particularly in https://github.com/envoyproxy/envoy/blob/master/source/common/common/fmt.h.

*Suggestion*:
Based on the significant changes in `fmt 5.0.0` a contributor with knowledge of `fmt` should incorporate the changes required to update. The current `fmt` release is 5.2.1 ([changelog](https://github.com/fmtlib/fmt/releases/tag/5.2.1)) and includes some performance updates.

`spdlog` releases follow `fmt` and the current release supporting `fmt 5.2.1` is `1.2.0` ([changelog](https://github.com/gabime/spdlog/releases/tag/v1.2.0)).

**Comment by zuercher** [comment]
I'll take a look at this.

**Comment by zuercher** [comment]
Closed in #4753
